### ISSUE TITLE
♻️ Remove `sanitizeString`

### ISSUE THREAD

**Issue by vd1992** [GCTC-NTGC/gc-digital-talent#11370]
## ♻️ Debt

Remove the `sanitizeString` function.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/970540aebe50ddfa0d33c440938b190c467ca1d1/packages/forms/src/utils.ts#L233

## 🕵️ Details

> [!IMPORTANT]
> There is a PHP function of the same name that should not be removed. 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/5996e758ea13fecba0b58ab63f95f2de0b31b5be/api/app/Traits/Generator/GeneratesFile.php#L76-L85

## 🙋‍♀️ Proposed Solution

Remove function `sanitizeString` and any implementations of it in the codebase.

## ✅ Acceptance Criteria

- [ ] no instances of JavaScript `sanitizeString` string in codebase

## 🛑 Blockers

```[tasklist]
### Blocked By
- [x] Meeting with IMTD about firewall
```

**Comment by petertgiles** [comment]
We met with IMTD about the firewall.  We haven't been able to get any data about hits that are happening or a list of firewall rules but they are very responsive to updating the rules when we report blocks happen due to certain characters.

My recommendation is to remove this function entirely.  It feels like an anitpattern to be proactively massaging user input to make the firewall happy unless there's something that we really can't adjust in the firewall and there's a clear sanitization operation that we can do to compensate.

**Comment by mnigh** [comment]
> My recommendation is to remove this function entirely. It feels like an anitpattern to be proactively massaging user input to make the firewall happy unless there's something that we really can't adjust in the firewall and there's a clear sanitization operation that we can do to compensate.

@petertgiles I agree with you. Does this mean that this issue can be rewritten to `Remove `sanitizeString`?

https://github.com/GCTC-NTGC/gc-digital-talent/blob/970540aebe50ddfa0d33c440938b190c467ca1d1/packages/forms/src/utils.ts#L233
